rust/agents/scraper/migration/src/m20220805_000004_create_table_message.rs

```diff
@@ -0,0 +1,125 @@
+                            .not_null(),
+                    )
+                    .foreign_key(
+                        ForeignKey::create()
+                            .from_col(Message::Origin)
+                            .to(Domain::Table, Domain::DomainId),
+                    )
+                    .foreign_key(
+                        ForeignKey::create()
+                            .from_col(Message::Destination)
+                            .to(Domain::Table, Domain::DomainId),
+                    )
+                    .foreign_key(
+                        ForeignKey::create()
+                            .from_col(Message::DispatchTxId)
+                            .to(Transaction::Table, Transaction::Id),
+                    )
+                    .index(
+                        Index::create()
+                            .unique()
+                            .col(Message::OutboxAddress)
+                            .col(Message::Origin)
+                            .col(Message::LeafIndex),
+                    )
+                    .to_owned(),
+            )
+            .await?;
+        manager
+            .create_index(
+                Index::create()
+                    .table(Message::Table)
+                    .name("idx-message_tx")
+                    .col(Message::DispatchTxId)
+                    .to_owned(),
+            )
+            .await?;
+        manager
+            .create_index(
+                Index::create()
+                    .table(Message::Table)
+                    .name("idx-message_sender")
+                    .col(Message::Sender)
+                    .to_owned(),
+            )
+            .await?;
+        manager
+            .create_index(
+                Index::create()
+                    .table(Message::Table)
+                    .name("idx-message_recipient")
+                    .col(Message::Recipient)
+                    .to_owned(),
+            )
+            .await?;
+
+        Ok(())
+    }
+
+    async fn down(&self, manager: &SchemaManager) -> Result<(), DbErr> {
+        manager
+            .drop_table(Table::drop().table(Message::Table).to_owned())
+            .await
+    }
+}
+
+/// Learn more at https://docs.rs/sea-query#iden
+#[derive(Iden)]
+pub enum Message {
+    Table,
+    /// Unique database ID
+    Id,
+    /// Time of record creation
+    TimeCreated,
+    /// Domain ID of the origin chain
+    Origin,
+    /// Domain ID of the destination chain
+    Destination,
+    /// Leaf index of this message in the merkle tree of the outbox
+    LeafIndex,
+    /// Address of the message sender on the origin chain (not necessarily the
+    /// transaction signer)
+    Sender,
+    /// Address of the message recipient on the destination chain.
+    Recipient,
+    /// Binary blob included in the message.
+    MsgBody,
+    /// Address of the outbox contract
+    OutboxAddress,
+    /// Transaction this message was sent in.
+    DispatchTxId,
```

Comment:
What do you think of renaming to `OriginTxId` and adding a `DestinationTxId` here?

Reply:
`OriginTxId` would be fine by me it makes a little less sense in my brain but it is more consistent with other parts of the system. `DestinationTxId` we can't know at the time the message was discovered so see previous comment.